### PR TITLE
Use TRAVIS_REPO_SLUG instead of hard-coding pytest-dev/pytest

### PR DESCRIPTION
**Pull request by nicoddemus**
I was doing final tests on the script today, and forgot to change
the hardecoded "pytest-dev/pytest", which ended up publishing a
`4.99.10` release to the main repository by mistake, as my token
has access to both my fork and main repository. 🤕 

I deleted the tag immeditely just a few seconds later, so hopefully
this won't cause major problems.

This change makes it safer to test this in the future, never publishing
to the main repository by mistake (as long as the tags are pushed to the
right repositories of course).

### CODE REVIEW

File: scripts/publish_gh_release_notes.py [hunk 1]
```diff
@@ -71,14 +72,19 @@ def main(argv):
         print("GH_RELEASE_NOTES_TOKEN not set", file=sys.stderr)
         return 1
 
+    slug = os.environ.get("TRAVIS_REPO_SLUG")
+    if not slug:
+        print("TRAVIS_REPO_SLUG not set", file=sys.stderr)
+        return 1
```
Comment (blueyed) [inline scripts/publish_gh_release_notes.py]: Just use `slug = os.environ["TRAVIS_REPO_SLUG"]` maybe to get a traceback also?  (i.e. more info)
But it's OK like that also.

File: scripts/publish_gh_release_notes.py [hunk 2]
```diff
@@ -71,14 +72,19 @@ def main(argv):
...
     rst_body = parse_changelog(tag_name)
     md_body = convert_rst_to_md(rst_body)
-    if not publish_github_release(token, tag_name, md_body):
+    if not publish_github_release(slug, token, tag_name, md_body):
         print("Could not publish release notes:", file=sys.stderr)
         print(md_body, file=sys.stderr)
         return 5
 
-    print(f"Release notes for {tag_name} published successfully")
+    print(f"Release notes for {tag_name} published successfully to {slug}")
```
Comment (blueyed) [inline scripts/publish_gh_release_notes.py]: Might be cool to have a link to it there, assuming it's in the response somehow.
(the next time you debug/change it ;))

File: scripts/publish_gh_release_notes.py (hunk 1 above)
Comment (nicoddemus) [reply scripts/publish_gh_release_notes.py]: I think the message is pretty clear TBH 😁

File: scripts/publish_gh_release_notes.py (hunk 2 above)
Comment (nicoddemus) [reply scripts/publish_gh_release_notes.py]: Great idea, done. 👍

File: scripts/publish_gh_release_notes.py (hunk 1 above)
Comment (blueyed) [reply scripts/publish_gh_release_notes.py]: Sure.
I've just meant that just asserting it to be there is enough/shorter, and provides validation (all in a single statement).

**Pull request merged**